openemrs/spec/requests/v1/doctors_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# doctors request spec
+
+require "rails_helper"
+
+RSpec.describe "V1::DoctorsController", type: :request do
+  let(:params) { attributes_for(:user) }
+
+  describe "#create" do
+    context "when user passes in valid doctor params" do
+      it "should create a doctor with a default role" do
+        expect do
+          post v1_doctors_path, params: params
+          returned_doctor = json["data"]["doctor"]
+          expect(response).to have_http_status 201
+          expect(returned_doctor["name"]).to eq params[:name]
+          expect(returned_doctor["email"]).to eq params[:email]
+          expect(returned_doctor["date_of_birth"]).to eq params[:date_of_birth].to_s
+          expect(returned_doctor["roles"]).to include "Doctor"
```

Comment:
You can dry this out by extending the last line here. For more information, checkout [expect to include](https://rubydoc.info/gems/rspec-expectations/RSpec/Matchers:include). You can build a hash like so

```
expected = {
  name: "John"
  email: "john.doe@example.com"
}
```

Then …

```
expect(returned_doctor).to include(**expected)
```

…or something like this. I am unsure if I used the double splat correctly. My ruby-fu has waned over the months